smoke-test/src/test/java/org/opennms/smoketest/ThresholdingIT.java

```diff
@@ -0,0 +1,263 @@
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.smoketest;
+
+import static com.google.common.io.Resources.getResource;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.not;
+import static org.opennms.netmgt.events.api.EventConstants.ABSOLUTE_CHANGE_THRESHOLD_EVENT_UEI;
+import static org.opennms.netmgt.events.api.EventConstants.HIGH_THRESHOLD_EVENT_UEI;
+
+import java.nio.file.attribute.PosixFilePermissions;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.Response;
+
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.opennms.core.utils.InetAddressUtils;
+import org.opennms.netmgt.model.OnmsAlarm;
+import org.opennms.netmgt.model.OnmsAlarmCollection;
+import org.opennms.netmgt.model.OnmsCategory;
+import org.opennms.netmgt.model.OnmsEvent;
+import org.opennms.netmgt.model.OnmsEventCollection;
+import org.opennms.netmgt.model.OnmsIpInterface;
+import org.opennms.netmgt.model.OnmsMetaData;
+import org.opennms.netmgt.model.OnmsMonitoredService;
+import org.opennms.netmgt.model.OnmsNode;
+import org.opennms.netmgt.model.OnmsServiceType;
+import org.opennms.netmgt.model.OnmsSeverity;
+import org.opennms.smoketest.stacks.OpenNMSProfile;
+import org.opennms.smoketest.stacks.OpenNMSStack;
+import org.opennms.smoketest.stacks.StackModel;
+import org.opennms.smoketest.utils.RestClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Tests to validate that thresholding works.
+ *
+ * @author jwhite
+ */
+public class ThresholdingIT {
+    private static final Logger LOG = LoggerFactory.getLogger(ThresholdingIT.class);
+
+    private static final String TEST_NODE_FS = "test";
+    private static final String TEST_NODE_FID = "thresholding-test-node";
+    private static final String TEST_NODE_CRITERIA = TEST_NODE_FS + ":" + TEST_NODE_FID;
+    private static final String TEST_NODE_CATEGORY = "ThresholdingTest";
+    private static final String TEST_SVC_NAME = "SvcToThreshold";
+
+    @ClassRule
+    public static final OpenNMSStack stack = OpenNMSStack.withModel(StackModel.newBuilder()
+            .withOpenNMS(OpenNMSProfile.newBuilder()
+                    .withFile(getResource("thresholding/poller-configuration.xml"), "etc/poller-configuration.xml")
```

Comment:
Is it worth trying to do the `getResouce(...)` call in `withFile(String, String)` instead as it is always performed anyways?

Reply:
Good point, this will be pretty common, so worth adding there.